tests/variational/test_inference.py

```
@@ -196,18 +197,27 @@ def test_fit_start(inference_spec, simple_model):
 
     # Minibatch data can't be extracted into the `observed_data` group in the final InferenceData
     [observed_value] = [simple_model.rvs_to_values[obs] for obs in simple_model.observed_RVs]
-    if observed_value.name.startswith("minibatch"):
```

What about something like

```python
expected_warning: bool = ...

with warnings.catch_warns(record=True) as record:
  ...
  try:
    ...
  except NotImplementedInference as err:
    ...
    
if expected_warning:
  assert len(record) == 1
  assert ... # Check expected warning
else:
  assert not record
```

We still want to raise for unexpected warnings

---

@ricardoV94 I added check for unexpected warnings